### ISSUE TITLE
Setting to not send usage statistics? 

### ISSUE THREAD

**Issue by finete** [CasperVerswijvelt/Better-Internet-Tiles#34]
According to fdroid this app sends usage statistics to a server.
Any way in the app to opt out of sending those? 

As this is an app with fairly high privileges.

**Comment by CasperVerswijvelt** [comment]
There is no way to disable the telemetry within the app. 

The code responsible for it is located [here](https://github.com/CasperVerswijvelt/Better-Internet-Tiles/blob/e37d78d27dc15e4cb1665a2bcb8f880e222a900e/app/src/main/java/be/casperverswijvelt/unifiedinternetqs/util/Util.kt#L377C5-L377C22). You could remove the content of that function and build the app yourself if you wish to use it without telemetry. That's not something I'm going to do, however, as I'd like to keep track a bit of what features are used most within the app.

The info that is sent out is fairly limited (some basic device information such as Android version, model and tile usage) and is only sent out a maximum of 1 time per 12 hours.

**Comment by D3SOX** [comment]
Forked the app and released latest version with some fixes and no telemetry under https://github.com/D3SOX/Better-Network-Tiles-Libre

**Comment by D3SOX** [comment]
I just noticed that you're actually still active on GitHub. My goal was not be a hostile fork. Would you accept a PR to disable all telemetry? I also did the German translations for v3 in my repo and fixed the notifications permission on Android 13+ :)

**Comment by CasperVerswijvelt** [comment]
As to why I'd like to keep telemetry/crashlytics enabled by default: it allows me to keep track on which tiles are being used most and, where I should focus on for future developments, as well as making me aware of crashes in the field. In addition, it's just fun to know how many people are actually using this app, across the multiple distribution platforms it is available on.
 
I do want to note that I use a fully open source simple analytics tool that I wrote myself, [report-anything](https://github.com/CasperVerswijvelt/report-anything), which only draws some basic graphs to interpret the data:


<details>
  <summary>Screenshot</summary>

  ![image](https://github.com/CasperVerswijvelt/Better-Internet-Tiles/assets/25928551/9c66c484-a9b0-4fed-800b-72c7d9208367)

</details>

**Comment by CasperVerswijvelt** [comment]
@D3SOX DId you actulaly experience any issues when the POST_NOTIFICATIONS permission was not present/requested?

The Android documentation states the following:

> Note: Apps don't need to request the POST_NOTIFICATIONS permission in order to launch a foreground service. However, apps must include a notification when they start a foreground service, just as they do on previous versions of Android.

https://developer.android.com/develop/ui/views/notifications/notification-permission

**Comment by D3SOX** [comment]
I added a permission request for it because otherwise the notifications permission was always off without manually enabling it

**Comment by CasperVerswijvelt** [comment]
> I added a permission request for it because otherwise the notifications permission was always off without manually enabling it on Android 13+

I don't think we actually need the notification permission at all. To start a foreground service you need to pass a notification, yes, but since Android 13 those are visualized differently using the new [Task Manager](https://source.android.com/docs/core/display/task-manager).

I never had any issues without the notification permission and the foreground service still worked fine
